patches/chrome-browser-ui-browser.cc.patch

```diff
@@ -0,0 +1,12 @@
+diff --git a/chrome/browser/ui/browser.cc b/chrome/browser/ui/browser.cc
+index bac1e524636c5046248c7fc3b4317ec8cb29574f..43a2466d3bcc144efaca7160d7141c4c7d35a4ec 100644
+--- a/chrome/browser/ui/browser.cc
++++ b/chrome/browser/ui/browser.cc
+@@ -466,6 +466,7 @@ Browser::Browser(const CreateParams& params)
+                           base::Unretained(this),
+                           BOOKMARK_BAR_STATE_CHANGE_PREF_CHANGE));
+ 
++  RegisterAlwaysShowBookmarkBarOnNTPChange
```

Comment:
There are some reason why I made a patch instead of subclassing `Browser` class.
1. `Browser` is created by static method `Browser::Create()`. So, we also have to change it to return `BraveBrowser`. This also causes patching to `browser.cc`.
2. To subclass, making a patch for `browser.h` is inevitable for overriding non-virtual method.
So, direct inserting it here is smallest patch.

Reply:
I don't understand, we have a patch to disable showing it on the NTP in bookmark_tab_helper.cc so now we're adding another patch to re-enable it?

Reply:
while this may be the smallest patch for this particular change, it is not very extensible. I think a better patch is a define in the header file which for now will just add a `friend` class. The friend class can have its own pref registrar and call `UpdateBookmarkBarState` using BrowserList (matching against the relevant profile). @iefremov already has a pref related BCKS in his network service PR that might make sense to reuse for this since a BCKS would make the most sense for this imo.

Reply:
BookmarkPrefsService is introduced.